tests/ui/associated-inherent-types/type-alias-bounds-are-enforced.rs

```diff
@@ -1,18 +1,12 @@
-// FIXME(inherent_associated_types): This should be `check-pass`
-// known-bug: #108491
 // compile-flags: --crate-type=lib
+// check-pass
```

Comment:
We no longer call `type_of` in variance computation on type aliases to determine if it's a TAIT, since we go through the `type_alias_is_lazy` query operating on HIR.